### PR TITLE
feat: include TypeScript types with release

### PR DESCRIPTION
**Pull request by JustinBeckwith**
**BREAKING CHANGE**: This commit switches many of the TypeScript interfaces from the types defined in `@types/request`, and replaces them with custom types included in the package.  This is only a breaking change at compile time for TypeScript users.

We currently rely on `@types/request`.  This causes a few issues:
- It gives users the impression we may include fields or properties we don't currently support
- We don't ship the types, so users just see `any`
- It requires pulling in `@types/request` for downstream users

I'd like to kinda drop all this by just shipping our own interfaces :)

### CODE REVIEW

**Comment by JustinBeckwith** [comment]
cc @ofrobots

File: src/index.ts [hunk 1]
```diff
@@ -10,16 +9,65 @@ export class RequestError extends Error {
   code?: number;
 }
 
-interface Headers {
-  [index: string]: string;
+export interface Headers {
+  // tslint:disable-next-line no-any
+  [index: string]: any;
```
Comment (ofrobots) [inline src/index.ts]: Why did you need to widen the type from `string` to `any`?

Note that this is semver major, but your PR is not marked as such.

File: package.json [hunk 2]
```diff
@@ -43,7 +43,6 @@
     "@types/mocha": "^5.2.5",
     "@types/nock": "^9.3.0",
     "@types/node-fetch": "^2.1.2",
-    "@types/request": "^2.47.1",
```
Comment (fhinkel) [inline package.json]: 👍

**Comment by fhinkel** [comment]
I'm not sure I'm 100% on board. If I understand correctly, users won't be able to use `teeny-request` as a drop in replacement for `request` but they'll have to change their type annotations?

**Comment by fhinkel** [comment]
I think we should ship the types, but keep this compatible with `request`.

**Comment by JustinBeckwith** [comment]
Ahhh - so just copy the `index.d.ts` from `@types/request`?  Just as an FYI, it pulls in a bunch of peer type dependencies as well :)

File: src/index.ts (hunk 1 above)
Comment (jkwlui) [reply src/index.ts]: https://nodejs.org/api/http.html#http_request_setheader_name_value

The header value is marked `any` in the Node docs.
Node typings defines `http.OutgoingHttpHeaders` as `number|string|string[]|undefined`;

Comment (jkwlui) [reply src/index.ts]: I see, you're mirroring `request` library's `Headers` type, which is in fact `[index: string]: any`

Comment (jkwlui) [reply src/index.ts]: `Headers` was used only internally (not exported), and I believe these new exported types are a 1:1 to the types in `r.Options`. Do we really need a semver major here?
